test/connection.test.ts

```diff
@@ -193,7 +193,143 @@ describe('Connection', function () {
+        const response = await realSendData.call(this.clientPlugin, data)
+        if(response[0] === IlpPacket.Type.TYPE_ILP_REJECT) {
+          rejected.push(IlpPacket.deserializeIlpReject(response))
+        }
+        return response
+      }
+
+      const clientStream = this.clientConn.createStream()
+      await clientStream.sendTotal(2000)
+
+      // F99X Are Application Errors due to the exchange rate dropping below the minimum acceptable amount
+      assert.equal(rejected[0].message, 'Packet amount too large')
+      assert.equal(rejected[1].code.includes('F99'), true)
+      assert.equal(rejected[2].code.includes('F99'), true)
+      assert.equal(rejected[3].code.includes('F99'), true)
+    })
+
+    it('should set slippage and modify the exchange rate until its worse than the minimum acceptable amount - slippage and the packets are rejected', async function () {      
+      this.clientPlugin.deregisterDataHandler()
+      this.serverPlugin.deregisterDataHandler()
+
+      this.server = await createServer({ //TODO: Move to createServer
+        plugin: this.serverPlugin,
+        serverSecret: Buffer.alloc(32)
+      })
+      await this.server.listen()
+
+      const { destinationAccount, sharedSecret } = this.server.generateAddressAndSecret()
+      this.destinationAccount = destinationAccount
+      this.sharedSecret = sharedSecret
+
+      const connectionPromise = this.server.acceptConnection()
+
+      this.clientConn = await createConnection({
+        plugin: this.clientPlugin,
+        destinationAccount,
+        sharedSecret,
+        slippage: 0.05
+      })
+      this.serverConn = await connectionPromise
+      this.serverConn.on('stream', (stream: DataAndMoneyStream) => {
+        stream.setReceiveMax(10000)
+      })
+
+      this.clientPlugin.maxAmount = 400
+      const exchangeRates = [0.5, 0.49, 0.48, 0.47, 0.5, 0.46, 0.7]
+      const realSendData = this.clientPlugin.sendData.bind(this.clientPlugin)
+      let callCount = 0
+      const args: Buffer[] = []
+      let rejected: Array<IlpPacket.IlpRejection> = []
+      this.clientPlugin.sendData = async (data: Buffer) => {
+        callCount++
+        args[callCount - 1] = data
+        if (callCount <= exchangeRates.length) {
+          this.clientPlugin.exchangeRate = exchangeRates[callCount - 1]
+        }
+        const response = await realSendData.call(this.clientPlugin, data)
+        if (response[0] === IlpPacket.Type.TYPE_ILP_REJECT) {
+          rejected.push(IlpPacket.deserializeIlpReject(response))
+        }
+        return response
+      }
+
+      const clientStream = this.clientConn.createStream()
+      await clientStream.sendTotal(2000)
+
+      // F99X Are Application Errors due to the exchange rate dropping below the minimum acceptable amount + slippage
+      assert.equal(rejected[0].message, 'Packet amount too large')
+      assert.equal(rejected[1].code.includes('F99'), true)
+      assert.equal(rejected[1].code.includes('F99'), true)
```

Comment:
Should this line be testing another call?

Reply:
Great catch, that should be the second rejected, updated the index to 2